### PR TITLE
[ar] Supervised Learning, ML tips and tricks

### PR DESCRIPTION
**Pull request by qunaieer**
[ar] Supervised Learning

### CODE REVIEW

**Comment by qunaieer** [comment]
I am planning to finish this file by the end of this week. I will merge it after I complete the file.

**Comment by qunaieer** [comment]
The translation of this file is complete, according to the terminologies on: https://www.nmthgiat.com/terminology/

**Comment by shervinea** [comment]
Thank you @qunaieer for all your work! Now, let's start the review process.

**Comment by zaidalyafeai** [comment]
Will work on the review. Thanks @qunaieer .

File: ar/cheatsheet-supervised-learning.md [hunk 1]
```diff
@@ -67,43 +67,43 @@
 
 **12. Loss function ― A loss function is a function L:(z,y)∈R×Y⟼L(z,y)∈R that takes as inputs the predicted value z corresponding to the real data value y and outputs how different they are. The common loss functions are summed up in the table below:**
 
-دالة الفرق (Loss function) - دالة الفرق هي الدالة L:(z,y)∈R×Y⟼L(z,y)∈R التي تأخذ كمدخلات القيمة المتوقعة z والقيمة الحقيقية y وتعطينا الفرق بينهما. الجدول التالي يحتوي على بعض دوال الفرق المستخدمة بكثرة:
+دالة الخسارة (Loss function) - دالة الخسارة هي الدالة L:(z,y)∈R×Y⟼L(z,y)∈R التي تأخذ كمدخلات القيمة المتوقعة z والقيمة الحقيقية y وتعطينا الفرق بينهما. الجدول التالي يحتوي على بعض دوال الخسارة الشائعة:
```
Comment (zaidalyafeai) [inline ar/cheatsheet-supervised-learning.md]: وتعطينا الإختلاف بينهما

File: ar/cheatsheet-supervised-learning.md [hunk 2]
```diff
@@ -67,43 +67,43 @@
...
 
-الهبوط التفاضلي (Gradient descent) - لنعرّف معدل التعلّم α∈R، يمكن تعريف القانون الذي يتم تحديث خوارزمية الهبوط التفاضلي من خلاله باستخدام معدل التعلّم ودالة التكلفة J كالتالي:
+النزول الاشتقاقي (Gradient descent) - لنعرّف معدل التعلّم α∈R، يمكن تعريف القانون الذي يتم تحديث خوارزمية النزول الاشتقاقي من خلاله باستخدام معدل التعلّم ودالة التكلفة J كالتالي:
 
 <br>
 
 **17. Remark: Stochastic gradient descent (SGD) is updating the parameter based on each training example, and batch gradient descent is on a batch of training examples.**
 
-ملاحظة: في الهبوط التفاضلي العشوائي (Stochastic gradient descent (SGD)) يتم تحديث المُعاملات (parameters) بناءاً على كل نقطة تدريب على حدة، بينما في الهبوط التفاضلي الحُزَمي (batch gradient descent) يتم تحديثها باستخدام حُزَم من نقاط التدريب.
+ملاحظة: في النزول الاشتقاقي العشوائي (Stochastic gradient descent (SGD)) يتم تحديث المُدخلات (parameters) بناءاً على كل عينة تدريب على حدة، بينما في النزول الاشتقاقي الحُزَمي (batch gradient descent) يتم تحديثها باستخدام حُزَم من عينات التدريب.
```
Comment (zaidalyafeai) [inline ar/cheatsheet-supervised-learning.md]: المدخلات يعني أنها قيم ثابتة تدخل بواسطة المستخدم بينما في الحقيقة هي تعتبر متغيرات

File: ar/cheatsheet-supervised-learning.md [hunk 3]
```diff
@@ -67,43 +67,43 @@
...
 
 <br>
 
 **18. Likelihood ― The likelihood of a model L(θ) given parameters θ is used to find the optimal parameters θ through maximizing the likelihood. In practice, we use the log-likelihood ℓ(θ)=log(L(θ)) which is easier to optimize. We have:**
 
-الأرجحية (Likelihood) - تستخدم أرجحية النموذج L(θ)، حيث أن θ هي المعاملات، للبحث عن أفضل المُعاملات θ عن طريق تعظيم (maximizing) الأرجحية. عملياً يتم استخدام الأرجحية اللوغاريثمية (log-likelihood) ℓ(θ)=log(L(θ)) حيث أنها أسهل في التحسين (optimize). فيكون لدينا:
+الأرجحية (Likelihood) - تستخدم أرجحية النموذج L(θ)، حيث أن θ هي المُدخلات، للبحث عن المُدخلات θ الأحسن عن طريق تعظيم (maximizing) الأرجحية. عملياً يتم استخدام الأرجحية اللوغاريثمية (log-likelihood) ℓ(θ)=log(L(θ)) حيث أنها أسهل في التحسين (optimize). فيكون لدينا:
```
Comment (zaidalyafeai) [inline ar/cheatsheet-supervised-learning.md]: الأحسن => الأمثل

File: ar/cheatsheet-supervised-learning.md [hunk 4]
```diff
@@ -139,31 +139,31 @@
 
 **24. Normal equations ― By noting X the matrix design, the value of θ that minimizes the cost function is a closed-form solution such that:**
 
-معادلة Normal - إذا كان لدينا المصفوفة X، القيمة θ التي تقلل من دالة التكلفة يمكن حلها رياضياً بشكل مغلق (closed-form) عن طريق:
+المعادلة الطبيعية/الناظمية (Normal) - إذا كان لدينا المصفوفة X، القيمة θ التي تقلل من دالة التكلفة يمكن حلها رياضياً بشكل مغلق (closed-form) عن طريق:
```
Comment (zaidalyafeai) [inline ar/cheatsheet-supervised-learning.md]: الطبيعية يعتبر ترجمة حرفية ، مكن استبدالها ب العمودية

File: ar/cheatsheet-supervised-learning.md [hunk 5]
```diff
@@ -277,31 +277,31 @@ Support Vector Machines
...
 
-النواة (Kernel) - إذا كان لدينا دالة تحويل الخصائص (features) ϕ، يمكننا تعريف النواة K كالتالي:
+النواة (Kernel) - إذا كان لدينا دالة ربط الخصائص (features) ϕ، يمكننا تعريف النواة K كالتالي:
 
 <br>
 
 **49. In practice, the kernel K defined by K(x,z)=exp(−||x−z||22σ2) is called the Gaussian kernel and is commonly used.**
 
-عملياً تُعَرَّف الدالة K عن طريق المعادلة K(x,z)=exp(−||x−z||22σ2)، ويطلق عليها النواة الجاوسية (Gaussian kernel)، وهي من الأكثر استخداماً.
+في التطبيق، يمكن أن تُعَرَّف الدالة K عن طريق المعادلة K(x,z)=exp(−||x−z||22σ2)، ويطلق عليها النواة الجاوسية (Gaussian kernel)، وهي تستخدم بكثرة.
```
Comment (zaidalyafeai) [inline ar/cheatsheet-supervised-learning.md]: في التطبيق => عملياً

**Comment by zaidalyafeai** [comment]
Finished half of the review @qunaieer .

**Comment by qunaieer** [comment]
Thank you @zaidalyafeai  
Please let me know if you finish, and I will go through the comments.

File: ar/cheatsheet-supervised-learning.md [hunk 6]
```diff
@@ -493,31 +493,31 @@ Probably Approximately Correct (PAC) - PAC هو إطار يتم من خلاله
 
 **83. Shattering ― Given a set S={x(1),...,x(d)}, and a set of classifiers H, we say that H shatters S if for any set of labels {y(1),...,y(d)}, we have:**
 
-Shattering - إذا كان لدينا المجموعة S={x(1),...,x(d)}، ومجموعة نماذج H، نقول أن H shatters S إذا كان لكل مجموعة أهداف (labels) {y(1),...,y(d)} لدينا:
+مجموعة تكسيرية (Shattering Set) - إذا كان لدينا المجموعة S={x(1),...,x(d)}، ومجموعة مُصنٍّفات H، نقول أن H shatters S إذا كان لكل مجموعة علامات (labels) {y(1),...,y(d)} لدينا:
```
Comment (zaidalyafeai) [inline ar/cheatsheet-supervised-learning.md]: ترجمة shatter ? 
اقترج تشطر أو تكسر

File: ar/cheatsheet-supervised-learning.md [hunk 7]
```diff
@@ -493,31 +493,31 @@ Probably Approximately Correct (PAC) - PAC هو إطار يتم من خلاله
...
 
-نظرية الحد الأعلى (Upper bound theorem) - لنجعل H فئة فرضية محدودة (finite hypothesis class) بحيث |H|=k، و δ وحجم العينة m ثابتين. حينها سيكون لدينا، مع احتمال على الأقل 1−δ، التالي:
+مبرهنة الحد الأعلى (Upper bound theorem) - لنجعل H فئة فرضية محدودة (finite hypothesis class) بحيث |H|=k، و δ وحجم العينة m ثابتين. حينها سيكون لدينا، مع احتمال على الأقل 1−δ، التالي:
 
 <br>
 
 **85. VC dimension ― The Vapnik-Chervonenkis (VC) dimension of a given infinite hypothesis class H, noted VC(H) is the size of the largest set that is shattered by H.**
 
-بُعد VC - بُعد فابنك-شيرفونينكز (Vapnik-Chervonenkis) لفئة فرضية محدودة (finite hypothesis class) H، ويرمز له بـ VC(H)، هو حجم أكبر مجموعة (set) التي shattered by H.
+بُعْد فابنيك-تشرفونيكس (Vapnik-Chervonenkis - VC) لفئة فرضية غير محدودة (infinite hypothesis class) H، ويرمز له بـ VC(H)، هو حجم أكبر مجموعة (set) التي shattered by H.
```
Comment (zaidalyafeai) [inline ar/cheatsheet-supervised-learning.md]: shattered by => تم تكسيرها بواسطة

**Comment by zaidalyafeai** [comment]
@qunaieer, I am done with the review.

**Comment by qunaieer** [comment]
Thank you @zaidalyafeai for the reivew.
How can I make changes? should I make them on the file in my machine, then do pull request? or is there a way to edit them directly here? @shervinea

**Comment by zaidalyafeai** [comment]
@qunaieer you are welcome. I think it is more convenient to make them offline then make a PR. You should use the same fork that you used initially in order to append the changes to the current PR.

File: ar/cheatsheet-supervised-learning.md (hunk 2 above)
Comment (qunaieer) [reply ar/cheatsheet-supervised-learning.md]: صحيح أتفق معك على الرغم من ترجمتها كذلك على الموقع، سأجعلها معامل على الرغم من أنها قد تشير إلى coefficient، ولكن ستفهم من السياق.

File: ar/cheatsheet-supervised-learning.md (hunk 3 above)
Comment (qunaieer) [reply ar/cheatsheet-supervised-learning.md]: سأبقيها الأحسن لاتفاق أغلب المترجمين عليها. الأمثل -- ideal.

File: ar/cheatsheet-supervised-learning.md (hunk 4 above)
Comment (qunaieer) [reply ar/cheatsheet-supervised-learning.md]: وجدناها كذلك في معاجم الرياضيات، فرجحنا المتفق عليه.

**Comment by qunaieer** [comment]
I have updated the file cheatsheet-supervised-learning.md according to @zaidalyafeai comments.

**Comment by shervinea** [comment]
Thank you very much @qunaieer and @zaidalyafeai for your hard work!
 
I see that the `machine-learning-tips-and-tricks` translation has also been added to this PR. @zaidalyafeai: could you please also review it? That way, we will be able to merge these two files at once.

**Comment by zaidalyafeai** [comment]
@shervinea sure, will take a stab at it asap.

File: ar/cheatsheet-machine-learning-tips-and-tricks.md [hunk 8]
```diff
@@ -1,285 +1,284 @@
...
 
-&#10230;
+[التدريب على k-1 جزء والتقييم باستخدام الجزء الباقي، التدريب على n−p عينة والتقييم باستخدام الـ p عينات المتبقية]
 
 <br>
 
 **31. [Generally k=5 or 10, Case p=1 is called leave-one-out]**
 
-&#10230;
+[بشكل غالب k=5 أو 10، الحالة p=1 يطلق عليها الإبقاء على واحد (leave-one-out)]
```
Comment (zaidalyafeai) [inline ar/cheatsheet-machine-learning-tips-and-tricks.md]: بشكل عام

**Comment by zaidalyafeai** [comment]
@qunaieer finished the review, excellent translation with minimal changes.

File: ar/cheatsheet-machine-learning-tips-and-tricks.md [hunk 9]
```diff
@@ -1,285 +1,284 @@
...
 
 <br>
 
 **32. The most commonly used method is called k-fold cross-validation and splits the training data into k folds to validate the model on one fold while training the model on the k−1 other folds, all of this k times. The error is then averaged over the k folds and is named cross-validation error.**
 
-&#10230;
+الطريقة الأكثر استخداماً يطلق عليها التحقق المتقاطع س جزء/أجزاء (k-fold)، ويتم فيها تقسيم البيانات إلى k جزء، بحيث يتم تدريب النموذج باستخدام k−1 والتحقق باستخدام الجزء المتبقي، ويتم تكرار ذلك k مرة. يتم بعد ذلك حساب معدل الأخطاء في الأجزاء k ويسمى خطأ التحقق المتقاطع.
```
[the remark this answers is not in the document]
Comment (qunaieer) [reply ar/cheatsheet-machine-learning-tips-and-tricks.md]: المقصود k-fold
وضعتها بين قوسين بعدها مباشرة للتوضيح مثل التالي:
س جزء/أجزاء (k-fold)

**Comment by qunaieer** [comment]
Thank you @zaidalyafeai for the review. I have changed the file accordingly and made a pull request.
I am not sure how to just select the changed file when doing pull request! It sends all the files I worked on previously.

**Comment by zaidalyafeai** [comment]
@qunaieer you can commit a certain file by `git commit -m "my message"  path/to/my/file`

**Comment by shervinea** [comment]
Thank you both @qunaieer and @zaidalyafeai for all your work. On my side, I added a few commits to:
- synchronize @qunaieer's fork to the repository's latest state
- restore a template that had been overwritten by the proposed merge and move the associated translation to the `[ar]` directory
- add your names in the contributors file

@zaidalyafeai: would you have any more comments or would you say that this PR is now ready for merge?

**Comment by qunaieer** [comment]
Thank you @shervinea 
As we discussed here https://github.com/shervinea/cheatsheet-translation/issues/84#issuecomment-544180410
It seems I need to add some tags to enable rtl. I will do this to the files and let you know.
Is this the method used for the other rtl languages (e.g., Persian)?

**Comment by shervinea** [comment]
@qunaieer Yes exactly, wrapping translations within `rtl` tags would be awesome (e.g. as in [#129](https://github.com/shervinea/cheatsheet-translation/pull/129/files)). Thank you for being so meticulous and for thinking of it!

**Comment by qunaieer** [comment]
Hello @shervinea I have updated the two files with RTL tags.

**Comment by shervinea** [comment]
Thank you very much @qunaieer, it looks great! Moving forward with the merge.

**Pull request merged**